hepstats/hypotests/calculators/frequentist_calculator.py

```diff
@@ -0,0 +1,430 @@
+                print("{0} toys generated, fitted and scanned!".format(i))
+
+            if i > ntoys:
+                break
+            i += 1
+
+        return result
+
+    def _gettoys(self, poigen, poieval=None, qtilde=False, hypotesis="null"):
+        """
+        Return the generated toys for a given POI.
+
+        Args:
+            poigen (POI): POI used to generate the toys
+            ntoys (int): number of toys to generate
+            poieval (POIarray): POI values to evaluate the loss function
+            qtilde (bool, optional): if `True` use the $$\tilde{q}$$ test statistics else (default) use
+                the $$q$$ test statistic
+            hypothesis: `null` or `alternative`
+        """
+
+        assert hypotesis in ["null", "alternative"]
+
+        if hypotesis == "null":
+            ntoys = self.ntoysnull
+            toysdict = self._toysnull
+        else:
+            ntoys = self.ntoysalt
+            toysdict = self._toysalt
+
+        for p in poigen:
+            if p not in toysdict:
+                ntogen = ntoys
+            else:
+                ngenerated = toysdict[p]["bestfit"]["values"].size
+                if ngenerated < ntoys:
+                    ntogen = ntoys - ngenerated
+                else:
+                    ntogen = 0
+
+            if ntogen > 0:
+                if self.printlevel >= 0:
+                    print(f"Generating {hypotesis} hypothesis toys for {p}.")
+
+                eval_values = [p.value]
+                if qtilde:
+                    eval_values.append(0.)
+                if poieval:
+                    eval_values += poieval.values.tolist()
+                eval_values = list(dict.fromkeys(eval_values))
+
+                toyresult = self._generate_and_fit_toys(p, ntogen, POIarray(poigen.parameter, eval_values))
+
+                if p in toysdict:
+                    toysdict[p].update(toyresult)
+                else:
+                    toysdict[p] = toyresult
+
+        return {p: toysdict[p] for p in poigen}
+
+    def gettoys_null(self, poigen, poieval=None, qtilde=False):
+        """
+        Return the generated toys for the null hypothesis.
+
+        Args:
+            poigen (POI): POI used to generate the toys
+            ntoys (int): number of toys to generate
+            poieval (POIarray): POI values to evaluate the loss function
+            qtilde (bool, optional): if `True` use the $$\tilde{q}$$ test statistics else (default) use
+                the $$q$$ test statistic
+
+        Example with `zfit`:
+            >>> mean = zfit.Parameter("mu", 1.2)
+            >>> poinull = POIarray(mean, [1.1, 1.2, 1.0])
+            >>> poialt = POI(mean, 1.2)
+            >>> for p in poinull:
+            ...     calc.gettoys_alt(p, poieval=poialt)
+        """
+        return self._gettoys(poigen=poigen, poieval=poieval, qtilde=qtilde, hypotesis="null")
+
+    def gettoys_alt(self, poigen, poieval=None, qtilde=False):
```

Comment:
I could have said it above already: personally I do not very much like `gettoys_XXX` and would have written `get_toys_XXX`. But up to you, of course.